### ISSUE TITLE
Flux Dreambooth/finetune 是否能在4090 24GVRAM上成功训练？

### ISSUE THREAD

**Issue by hotdogarea** [bmaltais/kohya_ss#2972]
我已经将batch_size设为最低1了，分辨率也是很正常的1024x1024 即便设为512x512 我还是收到了CUDA out of memory...
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 90.00 MiB. GPU 0 has a total capacity of 23.64 GiB of which 41.75 MiB is free. Including non-PyTorch memory, this process has 23.60 GiB memory in use. Of the allocated memory 23.19 GiB is allocated by PyTorch, and 5.97 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)

为了测试 我仅仅只放了四张图
![image](https://github.com/user-attachments/assets/559a35ce-bc00-4f50-8556-06ac6ece6565)

结果还是爆显存。底模使用的flux.1-dev  ，这是我的文件配置，我想知道通过怎么设置可以在24G显存的配置下对flux进行finetune训练
[test_20241118-061119.json](https://github.com/user-attachments/files/17797249/test_20241118-061119.json)

**Comment by hotdogarea** [comment]
> Full flux dev requires tons of vram and ram. You should either try "block to swap", or subscribe to him on Patreon for 24GB VRAM config. Have a great day :) 

Thank you very much, brother. The words woke me up. It is indeed this value. The larger the VRAM, the less need to set the swap value. After I set this value to 22, the training became normal.THX buddy！

**Comment by Nomination-NRB** [comment]
> > Full flux dev requires tons of vram and ram. You should either try "block to swap", or subscribe to him on Patreon for 24GB VRAM config. Have a great day :)
> 
> Thank you very much, brother. The words woke me up. It is indeed this value. The larger the VRAM, the less need to set the swap value. After I set this value to 22, the training became normal.THX buddy！

Hello, I also want to finetune the flux, and get the same error oom like you, could you please spare some time to help my issue:[https://github.com/bmaltais/kohya_ss/issues/3015](url)

**Comment by Sylsatra** [comment]
> > > Full flux dev requires tons of vram and ram. You should either try "block to swap", or subscribe to him on Patreon for 24GB VRAM config. Have a great day :)
> > 
> > 
> > Thank you very much, brother. The words woke me up. It is indeed this value. The larger the VRAM, the less need to set the swap value. After I set this value to 22, the training became normal.THX buddy！
> 
> Hello, I also want to finetune the flux, and get the same error oom like you, could you please spare some time to help my issue:[https://github.com/bmaltais/kohya_ss/issues/3015](url)

Hi, the link to your issue does not exist or it is closed!

**Comment by Nomination-NRB** [comment]
> > > > Full flux dev requires tons of vram and ram. You should either try "block to swap", or subscribe to him on Patreon for 24GB VRAM config. Have a great day :)
> > > 
> > > 
> > > Thank you very much, brother. The words woke me up. It is indeed this value. The larger the VRAM, the less need to set the swap value. After I set this value to 22, the training became normal.THX buddy！
> > 
> > 
> > Hello, I also want to finetune the flux, and get the same error oom like you, could you please spare some time to help my issue:[https://github.com/bmaltais/kohya_ss/issues/3015](url)
> 
> Hi, the link to your issue does not exist or it is closed!

Thanks for your reply, I update the link

**Comment by mimanchi96** [comment]
Hi, is finetuning for flux dev supported in kohyass training scripts? I don't have code knowledge and there is nothing about finetuning for flux dev on the web, looking forward to your reply. Thanks!

**Comment by Nomination-NRB** [comment]
> Hi, is finetuning for flux dev supported in kohyass training scripts? I don't have code knowledge and there is nothing about finetuning for flux dev on the web, looking forward to your reply. Thanks!

yes, if you meet oom problem, you should set block to swap according to your GPU VRAM, the larger value, the more vram save

**Comment by mimanchi96** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by hotdogarea** [comment]
24G VRAM has just set the block to swap=10, the GPU memory is exactly full, I will close this issue
